py/get_tiles.py

```diff
@@ -0,0 +1,349 @@
+        checklist.append(neighbor)
+        visited_tiles.add(neighbor)
+
+    return tilelist
+
+def check_args(argv):
+
+   global url, output_dir, boundingbox
+   try:
+      opts, args = getopt.getopt(sys.argv[1:], "h:b:u:d:", ["help=", "bbox=", "url=", "output_dir="])
+   except getopt.GetoptError:
+      print('tiles.py -b lower_left_lng_lat, upper_right_lng_lat -u https://thewebsite.com/dir -d /data/tiles')
+      print('tiles.py -b -74.251961,40.512764,-73.755405,40.903125 -u https://thewebsite.com/dir -d /data/tiles')
+      sys.exit(2)
+   for opt, arg in opts:
+      if opt in ("-h", "--help"):
+         print('tiles.py -b lower_left_lng_lat, upper_right_lng_lat -u https://thewebsite.com/dir -d /data/tiles')
+         print('tiles.py -b -74.251961,40.512764,-73.755405,40.903125 -u https://thewebsite.com/dir -d /data/tiles')
+         sys.exit()
+      elif opt in ("-b", "--bbox"):
+         boundingbox = arg
+      elif opt in ("-u", "--url"):
+         url = arg
+      elif opt in ("-d", "--output_dir"):
+         output_dir = arg
+
+   if (boundingbox == None or output_dir == None or url == None):
+      print('tiles.py -b lower_left_lng_lat, upper_right_lng_lat -u https://thewebsite.com/dir -d /data/tiles')
+      print('tiles.py -b -74.251961,40.512764,-73.755405,40.903125 -u https://thewebsite.com/dir -d /data/tiles')
+      sys.exit()
+
+#this is the entry point to the program
+if __name__ == "__main__":
+   
+  check_args(sys.argv[1:])
+
+  # these are the tiles that should exist in s3
+  tile_hierarchy = TileHierarchy()
+  for level, tiles in tile_hierarchy.levels.items():
+    tile_ids[level] = set()
+    for row in xrange(0, tiles.nrows):
+      for col in xrange(0, tiles.ncolumns):
+        tile_id = (row * tiles.ncolumns) + col
+        tile_ids[level].add(tile_id)
+
+  if boundingbox:
+    bbox = [ float(i) for i in boundingbox.split(',')]
+
+    bounding_boxes = []
+    #check our bb and make sure it does not cross 180/-180, if it does
+    #split it into two bounding boxes.
+    # example 174.223,-37.348,-175.314,-36.4099999 will change to
+    # 174.223,-37.348,180.0,-36.4099999 and
+    # -180.0,-37.348,-175.314,-36.4099999
+    if (bbox[0] >= bbox[2]) and (bbox[2] >= -180.0):
+      bounding_boxes.append(BoundingBox(bbox[0], bbox[1], 180.0, bbox[3]))
+      bounding_boxes.append(BoundingBox(-180.0, bbox[1], bbox[2], bbox[3]))
+    else:
+      bounding_boxes.append(BoundingBox(bbox[0], bbox[1], bbox[2], bbox[3]))
+
+    while (len(bounding_boxes) != 0):
+      b_box = bounding_boxes.pop(0)
+
+      for level, t_ids in tile_ids.items():
+        # only get the tiles that intersect the bounding box
+        tiles = tile_hierarchy.levels[level].TileList(b_box,t_ids)
+        for t in tiles:
+          file_name = tile_hierarchy.levels[level].GetFile(t, level)
+          print(url + file_name)
+          print("--create-dirs")
+          print("-o")
+          print(output_dir + file_name)
```

Comment:
i think this should be:

```python
print(file_name)
```

then change the bash script to be something like `xargs -I % curl http://some_s3_url/% --create-dirs -o %` this way we dont have the python spitting out anything but the tile names and we dont have to pass it a `url` or `outdir`